sample-app/src/main/java/at/bitfire/cert4android/demo/MainActivity.kt

```diff
@@ -179,10 +179,10 @@ class MainActivity : ComponentActivity() {
         private val _certificateFlow = MutableStateFlow<X509Certificate?>(null)
         val certificateFlow: StateFlow<X509Certificate?> = _certificateFlow
 
-        private val _userDecisionFlow = MutableStateFlow<Boolean?>(null)
+        private var userDecision: CompletableDeferred<Boolean> = CompletableDeferred()
```

Comment:
I'd also probably annotate it as volatile or use some kind of lock? I'm always really scared of vars for asynchronous code.

In any case, here it isn't really necessary, just an example implementation, and since everything is based on user interaction, I think there shouldn 't be any problems

Reply:
I think it makes sense to check for stuff like this. _Especially_ because it might be used as a reference implementation :+1: